File: java/std_nat/src/fan/std/TimePointPeer.java
package fan.std;

import fanx.main.Sys;
import fanx.main.Type;

public class TimePointPeer {

	private static Type type;

	public Type typeof() {
		if (type == null) {
			type = Sys.findType("std::TimeUtil");
		}
		return type;
	}

	static long nowMillis() {
		return System.currentTimeMillis();
	}

	static long nanoTicks() {
		return System.nanoTime();
	}

	static long nowUniqueLast = 0;

	static synchronized long nowUnique() {
		long now = System.currentTimeMillis();
		if (now <= nowUniqueLast)
			now = nowUniqueLast + 1;
		nowUniqueLast = now;
		return nowUniqueLast;
	}

}

File: src/std/js/TimePointPeer.js
fan.std.TimePointPeer = function(){}


fan.std.TimePointPeer.nowMillis = function() {
	var d = new Date();
	return d.getTime();
}

fan.std.TimePointPeer.nanoTicks = function() {
	var d = new Date();
	return d.getTime() * fan.std.Duration.m_nsPerMilli;
}

fan.std.TimePointPeer.nowUniqueLast = 0;

fan.std.TimePointPeer.nowUnique = function() {
	var d = new Date();
	var now = d.getTime();

	if (now <= fan.std.TimePointPeer.nowUniqueLast)
		now = fan.std.TimePointPeer.nowUniqueLast + 1;
	fan.std.TimePointPeer.nowUniqueLast = now;
	return now;
}
